corehq/apps/linked_domain/management/commands/add_app_id_to_linked_reports.py

```diff
@@ -0,0 +1,50 @@
+from django.core.management import BaseCommand, CommandError
+
+from corehq.apps.linked_domain.applications import get_downstream_app_id
+from corehq.apps.linked_domain.models import DomainLink
+from corehq.apps.userreports.dbaccessors import get_report_configs_for_domain
+from corehq.apps.userreports.models import DataSourceConfiguration, ReportConfiguration
+
+
+def migrate_linked_reports(upstream_domain=None):
+    if upstream_domain:
+        domain_links = DomainLink.objects.filter(master_domain=upstream_domain)
+    else:
+        domain_links = DomainLink.objects.filter(deleted=False)
+
+    for domain_link in domain_links:
+        reports = get_report_configs_for_domain(domain_link.linked_domain)
+        for report in reports:
+            if report.report_meta.master_id and not report.config.meta.build.app_id:
+                upstream_report = ReportConfiguration.get(report.report_meta.master_id)
+                upstream_datasource = DataSourceConfiguration.get(upstream_report.config_id)
+                downstream_app_id = get_downstream_app_id(
+                    domain_link.linked_domain,
+                    upstream_datasource.meta.build.app_id,
+                )
+                if not downstream_app_id:
+                    # just as a backup in case upstream_app_id is not set but family_id is
+                    downstream_app_id = get_downstream_app_id(
+                        domain_link.linked_domain,
+                        upstream_datasource.meta.build.app_id,
+                        use_upstream_app_id=False
+                    )
+                if not downstream_app_id:
+                    raise CommandError(f"Could not find downstream_app_id for upstream app"
```

Comment:
Does this signify a problem with the script or is it expected, say, if an app has been deleted? If it's the latter, you probably want to write to a log instead of failing the script.

Reply:
Ah writing to a log makes sense. I wasn't sure how to handle this so that I could be aware of which ids failed, but able to continue with the script. Now that you say that, writing to a log is the obvious choice. Thanks.